core/app/models/spree/order_updater.rb

```diff
@@ -77,15 +77,14 @@ def update_order_total
 
     def update_adjustment_total
       recalculate_adjustments
-      order.adjustment_total = line_items.sum(:adjustment_total) +
-      shipments.sum(:adjustment_total) +
-      adjustments.eligible.sum(:amount)
-      order.included_tax_total = line_items.sum(:included_tax_total) + shipments.sum(:included_tax_total)
-      order.additional_tax_total = line_items.sum(:additional_tax_total) + shipments.sum(:additional_tax_total)
 
-      order.promo_total = line_items.sum(:promo_total) +
-      shipments.sum(:promo_total) +
-      adjustments.promotion.eligible.sum(:amount)
+      all_items = line_items + shipments
+
+      order.adjustment_total = all_items.sum(&:adjustment_total) + adjustments.eligible.sum(:amount)
+      order.included_tax_total = all_items.sum(&:included_tax_total)
+      order.additional_tax_total = all_items.sum(&:additional_tax_total)
+
+      order.promo_total = all_items.sum(&:promo_total) + adjustments.promotion.eligible.sum(:amount)
 
       update_order_total
     end
```

Comment:
That's all pretty great! Thank you for it!

Just wanted to ask you on why you didn't go about to change `adjustments.eligible` to an in-memory call as well? I think the collection has already been loaded by now somewhere in the `ItemAdjustments`, and here you're calling a sum twice, so it should be a two SQL lookup save. A `#promotion?` method already exists on Adjustment, only an `#eligible?` one would need be added, which could be useful.